mantidproject/mantid
# Logs and stats can be filtered or unfiltered by a checkbox

Adds a filtered checkbox to the workbench sample logs interface to allow filtering or the whole data to be shown.

**To test:**
1, in workbench load POLREF00014966 (unit test)
1. click ws->show sample logs
1. The GUI has a new checkbox at the top "Filtered data"  This filters both the plot and the stats.
1. Go through some of the logs, logs without a filter will disable the checkbox. (Generally, only double time series logs are filtered).
1. For multiple selections the box will be enabled if any log is filtered.
1. Pick a log like raw_uah_log, flick between filtered and unfiltered, see the log and the plots change.
1. Check that if you double click the plot it extracts a matching filtered plot.
1. Check that filtering on the plot works if you have multiple selections (add good_uah_log with ctrl click)

re #28569


---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
